cognite/client/data_classes/workflows.py

```diff
@@ -0,0 +1,928 @@
+from __future__ import annotations
+
+import json
+from abc import ABC, abstractmethod
+from collections import UserList
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any, ClassVar, Literal, Sequence, cast
+
+from typing_extensions import Self, TypeAlias
+
+from cognite.client.data_classes._base import (
+    CogniteResource,
+    CogniteResourceList,
+    T_CogniteResource,
+)
+from cognite.client.utils._text import convert_all_keys_to_snake_case, to_snake_case
+
+if TYPE_CHECKING:
+    from cognite.client import CogniteClient
+
+WorkflowStatus: TypeAlias = Literal[
+    "in_progress",
+    "cancelled",
+    "failed",
+    "failed_with_terminal_error",
+    "completed",
+    "completed_with_errors",
+    "timed_out",
+    "skipped",
+]
+
+
+class WorkflowCreate(CogniteResource):
+    """
+    This class represents a workflow. This is the write version, used when creating or updating a workflow.
+
+    Args:
+        external_id (str): The external ID provided by the client. Must be unique for the resource type.
+        description (str | None): Description of the workflow. Defaults to None.
+
+    """
+
+    def __init__(self, external_id: str, description: str | None = None) -> None:
```

Comment:
having default value of `None` here for description is sketchy in the context of an upsert. It's easy to inadvertently wipe the description then. I think we should not have default values when the only upsert mode we support is `replace`.